doc/Create-resources-using-aso.md

```diff
@@ -0,0 +1,200 @@
+```
+4. Deploy the AZURE secret:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: aso-credential
+ namespace: default
+stringData:
+ AZURE_SUBSCRIPTION_ID: "$AZURE_SUBSCRIPTION_ID"
+ AZURE_TENANT_ID: "$AZURE_TENANT_ID"
+ AZURE_CLIENT_ID: "$AZURE_CLIENT_ID"
+ AZURE_CLIENT_SECRET: "$AZURE_CLIENT_SECRET"
+EOF
+```
+## Crating infrastructure for ARO-HCP clusters
+5. Then we can start such a deplouyment:
+```bash
+kubectl -f - <<EOF
+# az group create --name <resource-group> --location <location>
+apiVersion: resources.azure.com/v1api20200601
+kind: ResourceGroup
+metadata:
+  name: mv-rg
+  namespace: default
+spec:
+  location: eastus
+---
+# az network vnet create -n <vnet-name> -g <resource-group> --subnet-name <subnet-name>
+apiVersion: network.azure.com/v1api20201101
+kind: VirtualNetwork
+metadata:
+  name: mv-vnet
+  namespace: default
+spec:
+  location: eastus
+  owner:
+    name: mv-rg
+  addressSpace:
+    addressPrefixes:
+      - 10.100.0.0/15
+---
+# az network nsg create -n <nsg-name> -g <resource-group>
+apiVersion: network.azure.com/v1api20201101
+kind: NetworkSecurityGroup
+metadata:
+  name: mv-sg
+  namespace: default
+spec:
+  location: eastus
+  owner:
+    name: mv-rg
+---
+# az network vnet create -n <vnet-name> -g <resource-group> --subnet-name <subnet-name>
+# az network vnet subnet update -g <resource-group> -n <subnet-name> --vnet-name <vnet-name> --network-security-group <nsg-name>
+apiVersion: network.azure.com/v1api20201101
+kind: VirtualNetworksSubnet
+metadata:
+  name: mv-subnet
+  namespace: default
+spec:
+  owner:
+    name: mv-vnet
+  addressPrefix: 10.100.76.0/24
+  networkSecurityGroup: 
+    reference:
+      name: mv-sg
+      group: network.azure.com
+      kind: NetworkSecurityGroup
+EOF
+USER=mv
+CS_CLUSTER_NAME=cluster1
+OPERATORS_UAMIS_SUFFIX=$(openssl rand -hex 3)
+for IDENTITY_NAME in \
+    ${USER}-${CS_CLUSTER_NAME}-cp-control-plane-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-cluster-api-azure-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-cloud-controller-manager-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-ingress-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-disk-csi-driver-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-file-csi-driver-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-image-registry-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-cloud-network-config-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-cp-kms-${OPERATORS_UAMIS_SUFFIX} \
+    \
+    ${USER}-${CS_CLUSTER_NAME}-dp-disk-csi-driver-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-dp-image-registry-${OPERATORS_UAMIS_SUFFIX} \
+    ${USER}-${CS_CLUSTER_NAME}-dp-file-csi-driver-${OPERATORS_UAMIS_SUFFIX} \
+    \
+    ${USER}-${CS_CLUSTER_NAME}-service-managed-identity-${OPERATORS_UAMIS_SUFFIX} \
+; do 
+kubectl apply -f - <<EOF
```

Comment:
better to push the output of the for loop to a file ex; AroHcpUserAssignedIdentity.yaml then
 `oc apply -f AroHcpUserAssignedIdentity.yaml`